test/libdnf/utils.hpp

```diff
@@ -155,6 +223,8 @@ struct assertion_traits<libdnf::base::TransactionPackage> {
 
 
 std::vector<libdnf::advisory::Advisory> to_vector(const libdnf::advisory::AdvisorySet & advisory_set);
+std::vector<libdnf::comps::Environment> to_vector(const libdnf::Set<libdnf::comps::Environment> & environment_set);
+std::vector<libdnf::comps::Group> to_vector(const libdnf::Set<libdnf::comps::Group> & group_set);
```

Comment:
I use `libdnf::Set` for sets of `Groups`/`Environments`, but I don't know if it's correct - if it shouldn't rather be `std::set` (but then it couldn't be easily converted from `GroupQuery`) or the query class (that would be probably too restrictive?), and why there is for example `AdvisorySet` as a special class instead of having just `std::set<Advisory>` or `libdnf::Set<Advisory>`.

Reply:
Using `libdnf::Set` is correct at this stage of things AFAIU. The reason `AdvisorySet` is a special class is because it uses `SolvMap` as its underlying storage (same as PackageSet), which is more effective at least for when you need to pass the `SolvMap` to libsolv (the fact that `SolvMap` is a bitmap has other advantages and also some disadvantages). Your items aren't represented by a single solvable Id so you can't use that.

As for why not just use `std::set` (which is actually the underlying storage for `libdnf::Set`), it's just to keep a consistent interface with the `SolvMap`-based sets. I'm not sure how important is that and maybe we should consider if we should just use `std::set` for simplicity. Not sure atm.